### PR TITLE
MRG: Allow setting mag_scale in maxwell_filter

### PR DESCRIPTION
**Pull request by Eric89GXL**
From talking to @staulu this seems like a better / more principled way to set `mag_scale` in Maxwell filtering. This should help remove some of the mystery from the 100 value that used to be there.

@wronk can you have a look and see if it makes sense to you?

### CODE REVIEW

File: mne/preprocessing/maxwell.py [hunk 1]
```diff
@@ -503,6 +518,36 @@ def maxwell_filter(raw, origin='auto', int_order=8, ext_order=3,
...
+                             % mag_scale)
+        if len(mag_picks) in (0, len(meg_picks)):
+            mag_scale = 100.  # only one coil type, doesn't matter
+            logger.info('    Setting mag_scale=%0.2f because only one '
+                        'coil type is present' % mag_scale)
+        else:
+            # Find our gradiometer baseline
+            coils = _create_meg_coils(pick_info(info, meg_picks)['chs'],
+                                      'accurate')
+            grad_base = set(coils[pick]['base'] for pick in grad_picks)
```
Comment (Eric89GXL) [inline mne/preprocessing/maxwell.py]: physical size of the sensor (where the difference is taken essentially)

Comment (wronk) [inline mne/preprocessing/maxwell.py]: Do you mind modifying the `# Find our gradiometer baseline` line to indicate that it's related to the physical sensor size? Also, Matti refers to this as the "base line". Maybe something like:
`# Find our gradiometer "base line" (physical distance between pickup loops)`

Other than that, LGTM

File: mne/epochs.py [hunk 2]
```diff
@@ -2979,6 +2979,15 @@ def average_movements(epochs, head_pos=None, orig_sfreq=None, picks=None,
         with reference channels is not currently supported.
     return_mapping : bool
         If True, return the mapping matrix.
+    mag_scale : float | str
+        The magenetometer scale-factor used to bring the magnetometers
+        to approximately the same order of magnitude as the gradiometers
+        (default 100.), as they have different units (T vs T/m).
+        Can be ``'auto'`` to use the reciprocal of the gradiometer
+        baseline (e.g., 0.0168 m yields 59.5 for VectorView).
```
Comment (wronk) [inline mne/epochs.py]: Here too

File: mne/preprocessing/maxwell.py [hunk 3]
```diff
@@ -136,6 +137,15 @@ def maxwell_filter(raw, origin='auto', int_order=8, ext_order=3,
 
         .. versionadded:: 0.12
 
+    mag_scale : float | str
+        The magenetometer scale-factor used to bring the magnetometers
+        to approximately the same order of magnitude as the gradiometers
+        (default 100.), as they have different units (T vs T/m).
+        Can be ``'auto'`` to use the reciprocal of the gradiometer
+        baseline (e.g., 0.0168 m yields 59.5 for VectorView).
```
Comment (wronk) [inline mne/preprocessing/maxwell.py]: And here

**Comment by Eric89GXL** [comment]
Done thanks @wronk

**Comment by Eric89GXL** [comment]
@agramfort okay to merge?

**Comment by agramfort** [comment]
yes

**Pull request merged**